src/oneD/Sim1D.cpp

```diff
@@ -445,14 +445,21 @@ int Sim1D::setFixedTemperature(double t)
         size_t npnow = d.nPoints();
         size_t nstart = znew.size();
         if (d_free && d_free->domainType() == cFreeFlow) {
+            double ttol = 1.e-3; // avoid new point too close to existing point
             for (size_t m = 0; m < npnow-1; m++) {
-                if (value(n,2,m) == t) {
+                if (fabs(t - value(n, 2, m)) <= ttol * t) {
                     zfixed = d.grid(m);
                     d_free->m_zfixed = zfixed;
                     d_free->m_tfixed = t;
                     addnewpt = false;
                     break;
-                } else if ((value(n,2,m)<t) && (value(n,2,m+1)>t)) {
+                } else if (fabs(value(n, 2, m + 1) - t) <= ttol * t) {
+                    zfixed = d.grid(m + 1);
+                    d_free->m_zfixed = zfixed;
+                    d_free->m_tfixed = t;
+                    addnewpt = false;
+                    break;
```

Comment:
I'm not sure I understand the purpose of adding this second case - won't point `m+1` always be considered in the next iteration of the loop over `m`?

Reply:
It would, but there is one more point to check than there are iterations, and end points need to be checked. I didn’t feel like it was a speed-critical issue.

Reply:
I'm more concerned about the code duplication than the redundant computation.

In addition, I don't think the last grid point should be a candidate for the fixed point temperature, given that the last grid point is usually being used to impose a zero gradient condition.

Reply:
I agree that the last point isn’t a good candidate, but I’m not sure that it is safe to assume that it won’t come up. Personally I thought that it was straightforward from a conceptual viewpoint (code readability) ... I.e. you check whether a new point would be close to the ‘left’ and ‘right’ point within each interval.

Reply:
PS: one thing that I did consider here would be to prevent closely spaced grid-points (which would be a change of behavior, but probably advantageous from a numerical point of view). I.e. if we’re close to a point, just adopt it and update the temperature ... thoughts?

Reply:
I streamlined the code to take care of code duplications.